src/development/troubleshoot.md

```diff
@@ -161,3 +135,75 @@ where `<project_id>` is the random-character ID of the project.  That can be fou
+3. The deployment is blocked by a long running cron job in the environment.
+4. The deployment is blocked by a long running cron job in the parent environment.
+
+To determine if your environment is being stuck in the build or the deployment, you should look at the build log available on the UI. When you see a log line like below, your application is built and the deployment is stuck:
+
+```
+Re-deploying environment w6ikvtghgyuty-drupal8-b3dsina.
+```
+
+Here are the things you should do when you see your build or deployment is stuck:
+
+For a blocked _build_ (when you don't find the `Re-deployment environment ...` line), you have to create a [support ticket](https://platform.sh/support) to let us kill it.
+
+When a _deployment_ is blocked, you should try the following:
+
+1. Use [SSH](/development/access-site.md) to connect to your environment. Find any long-running cron jobs on the environment by running `ps afx`. Once you have identified the long running process on the environment, kill it.
+2. If you're performing "Sync", "Merge", or "Activate" on an environment and the process is stuck, use [SSH](/development/access-site.md) to connect to the parent environment and identify any long running cron jobs with `ps afx`. Kill the job(s) if you see any.
+
+Stuck builds should automaticly be killed after one hour. If they are stuck for a longer period please create [support ticket](https://platform.sh/support).
+
+
+## Slow or failing build (or deployment)
+
+Builds that take long time or fail is a common problem. Most of the time it's related to an application issue and they can be hard to troubleshoot without guidance.
+
+Here are a few tips that can help you solve the issues you are experiencing.
+
+### Check for errors in the logs
+
+Invisible errors during the build and deploy phase can cause increased wait times, failed builds and other problems. Investigating each log and fixing errors is essential.
+
+Related documentation: [Accessing logs](https://docs.platform.sh/development/logs.html#accessing-logs)
+
+### Check php.access.log
+
+The php.access.log contains the page executions by php-fpm. It also includes the execution time and peak memory usage of each request. This can be used to find pages and scripts that use too much memory or time to finish.
+
+Show 10 slowest page loads in the last 1000 requests: `tail -n 1000 php.access.log | sort -n -k 4 | tail`
+
+### Build and deploy hooks
+
+Hooks are almost always the cause of long build time. If they run into problem they can cause the build to fail or hang indefinitely.
+
+You should test each hook and deploy command in your local development environment.
+
+Furthermore, you can test your hooks with those linux commands to help figure out any problems
+
+```
+time $cmd # Print execution time
+strace -T $cmd # Print a system call report
+```
+
+Related documentation: [Build and deploy hooks](https://docs.platform.sh/configuration/app/build.html#hooks)
```

Comment:
Somewhere in this section we should note that rerunning deploy hooks on an active site is *usually* safe, at least from our end, so debugging it there is fine.  (Modulo cache clearing and the impact of that, etc., which we should mention.)

Build hooks have to be debugged locally, but you can also see how long they take on push because the log is streamed in real-time.  So you can see if composer just takes forever or npm does or something else.

(This is why build and deploy should get their own sections.)

Reply:
@ralt Mentioned it was impossible due to the fact that the environment is read-only, so it needs to be tested locally.